modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4124,12 +4142,28 @@ def compute_groupby(df, drop=False, partition_idx=0):
         else:
             apply_indices = None
 
+        if (
+            agg_kwargs.get("as_index", True)
+            and len(not_broadcastable_by) == 0
+            and len(broadcastable_by) == 1
```

Comment:
Why such restrictions?

Reply:
These restrictions describe a simple case when 'by' are columns from the same dataframe. I don't want to handle other cases in this PR as pandas is always tricky in how it handles groupby with mixed `by`

Reply:
added an in-code comment about this